### ISSUE TITLE
dataloader memory leak issue

### ISSUE THREAD

**Issue by wuqianliang** [rabbityl/lepard#12]
Hi, thank you for your excellent work.
When I run your code, it seems always to have the data loader worker killed after running for a while. Is there a possible memory leak? 
I have set the load workers to 4 and batch size to 2 and have a TITAN card and 72G memory.
![image](https://user-images.githubusercontent.com/22786710/165093935-8d351cdc-09eb-4bb4-a16f-7bf49b93da4f.png)

**Comment by rabbityl** [comment]
I have never seen this on my machine. 
It's possibly caused by memory size.
Maybe you can try to use  a smaller num_worker, say 1, 2 or even 0.  
also see this: https://github.com/pytorch/pytorch/issues/8976#issuecomment-481442240

**Comment by chalth** [comment]
Hi， I meet the same problem, did you slove it?

**Comment by ACuOoOoO** [comment]
I find that the problem is caused by AverageMeter.update() in a training step. I solve the problem by detach the input tensor during the accumulation in AverageMeter.update().
